crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -469,16 +469,23 @@ impl<'db> TypeInferenceBuilder<'db> {
         let class_definitions = self
             .types
             .declarations
-            .values()
-            .filter_map(|ty| ty.into_class_literal())
-            .map(|class_ty| class_ty.class);
+            .iter()
+            .filter_map(|(definition, ty)| {
+                // Filter out class literals that result from imports
+                if matches!(definition.kind(self.db), DefinitionKind::Class(_)) {
+                    ty.into_class_literal()
+                } else {
+                    None
+                }
+            })
+            .map(|class_ty| (class_ty.class, class_ty.class.node(self.db)));
```

Comment:
Sorry, what I meant is this

```suggestion
            .iter()
            .filter_map(|(definition, ty)| {
                // Filter out class literals that result from imports
                if let DefinitionKind::Class(class) = definition.kind(self.db) {
                    ty.into_class_literal().map(|ty| (class, ty.class))
                } else {
                    None
                }
            });
```

Reply:
Ah, okay. Now it makes more sense. I assume you mean `class.node()` instead of `class`?